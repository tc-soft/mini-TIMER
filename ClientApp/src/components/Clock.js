import React from "react";

function Clocik({ className = "", minutes = 20, seconds = 48 }) {
    return <h2 className={"Clocik " + className}>Pozostało {minutes}:{seconds}</h2>
}

export default Clocik;
