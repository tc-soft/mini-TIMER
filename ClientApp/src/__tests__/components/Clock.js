import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer"
import Clock from "../../components/Clock";

var root = null;
var clockRenderer = null;
describe("<Clock />", () => {

    describe("when given minutes and seconds (TEST DOM)", () => {
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
                <Clock minutes={10} seconds={20} />, root
            )
        })

        it("renders an h2 element", () => {
            expect(root.childNodes[0].nodeName).toEqual("H2")
        })

        it("renders a Clock className", () => {
            expect(root.childNodes[0].className).toMatch(/Clock/)
        })

        it("renders time properly", () => {
            expect(root.childNodes[0].textContent).toMatch(/10:20/)
        })

        
    });

    describe("when given minutes and seconds (TestRenderer)", () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={10} seconds={20} />
            );
            
        })

        it("renders an h2 element", () => {
            expect(clockRenderer.toJSON()).toMatchSnapshot();
        })

        it("renders an h2 element", () => {
            expect(clockRenderer.toJSON().type).toEqual("h2")
        })

        it("sets a Clock className", () => {
            expect(clockRenderer.toJSON().props).toMatchObject({ "className": expect.stringMatching(/Clock/) })
        })

        it("renders time properly", () => {
            expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["10", "20"]))
        })

    });



})