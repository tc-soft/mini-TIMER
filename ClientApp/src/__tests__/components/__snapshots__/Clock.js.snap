// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (TestRenderer) renders an h2 element 1`] = `
<h2
  className="Clock "
>
  Pozosta≈Ço 
  10
  :
  20
</h2>
`;
